ark/internal/controller/a2aserver_controller.go

```diff
@@ -149,46 +148,42 @@ func (r *A2AServerReconciler) reconcileConditionsAddressResolutionFailed(ctx con
 	return nil
 }
 
-// reconcileConditionsDiscoveryFailed updates conditions and emits events when discovery fails
+// reconcileConditionsDiscoveryFailed updates conditions when discovery fails
 func (r *A2AServerReconciler) reconcileConditionsDiscoveryFailed(ctx context.Context, a2aServer *arkv1prealpha1.A2AServer, err error, resolvedAddress string) error {
 	log := logf.FromContext(ctx)
 	changed := r.reconcileCondition(a2aServer, A2AServerReady, metav1.ConditionFalse, "DiscoveryFailed", fmt.Sprintf("Server not ready due to discovery failure: %v", err))
 	if changed {
 		log.Error(err, "A2A agent discovery failed", "server", a2aServer.Name, "address", resolvedAddress)
-		r.Recorder.Event(a2aServer, corev1.EventTypeWarning, "AgentDiscoveryFailed", fmt.Sprintf("Failed to discover agents from A2A server %s: %v", resolvedAddress, err))
 		return r.updateStatusWithConditions(ctx, a2aServer)
 	}
 	return nil
 }
 
-// reconcileConditionsAgentCreationFailed updates conditions and emits events when agent creation fails
+// reconcileConditionsAgentCreationFailed updates conditions when agent creation fails
 func (r *A2AServerReconciler) reconcileConditionsAgentCreationFailed(ctx context.Context, a2aServer *arkv1prealpha1.A2AServer, err error, agentName string) error {
 	log := logf.FromContext(ctx)
 	changed := r.reconcileCondition(a2aServer, A2AServerReady, metav1.ConditionFalse, "AgentCreationFailed", fmt.Sprintf("Failed to create agent: %v", err))
 	if changed {
 		log.Error(err, "A2A agent creation failed", "server", a2aServer.Name, "agent", agentName)
```

Comment:
curious here - do we need to log at all in scenarios like this, or would the eventing call be sufficient? if in eventing we have a debug log or something we could eliminate a log of `log.Error` lines which in 99% of cases seem redundant. Also it might be best in the below to be clear it is an A2A agent